### ISSUE TITLE
Triggerings Liste leer obwohl RigidBody in einem ist

### ISSUE THREAD

**Issue by Jonathan-Schnee** [JirkaDellOro/FUDGE#293]
![grafik](https://user-images.githubusercontent.com/48802338/153862810-1867d12e-7d8a-4607-af38-2ed5a09bf4ff.png)

Hier sieht man das der agentRB 2 collisions hat.

![grafik](https://user-images.githubusercontent.com/48802338/153862822-86405cd5-3301-4771-b044-8ab157ba1993.png)

Hier sieht man das einer der beiden isTrigger auf true ist

![grafik](https://user-images.githubusercontent.com/48802338/153862858-998dfd66-dc9b-4e9b-9094-f5d072259e00.png)

Aber trotzdem ist die triggerings Liste leer bei dem AgentRB

**Comment by JirkaDellOro** [comment]
Please post links to your code and to the application. Maybe Marko wants to look into this

**Comment by Jonathan-Schnee** [comment]
Code: https://github.com/Jonathan-Schnee/RougeDungeon/tree/main/RougeDungeon
Application: https://jonathan-schnee.github.io/RougeDungeon/RougeDungeon/index.html

**Comment by Dev-MarkoF** [comment]
Hello, @Jonathan-Schnee 
first off, sorry i did not see this sooner, github did not show me a notification until today.

You found a way around it? Your application seems to work the way you want it.


But on the other hand:
I see multiple problems while browsing through your code.
You do not have a Event Handling for Triggers, so there is no callback when you enter a trigger.

**TL;DR;**
For reference look at [Physic Events Tutorial](https://github.com/JirkaDellOro/FUDGE/tree/development/Documentation/Tutorials/Physics/FinishedExamples/Lesson_3_Physical%20Events)

**HELPFUL LONG ANSWER**

I did not look through all your code, but you are missing the basic structure of,
1. Have a Rigidbody that is an actual physics body (so not a trigger, your player for example)
2. A `Event Caller` on either the trigger or the physics body 
3. A `Event Handler` that is registering the triggering and does something in your case it seems hack the tree

Currently it seems you do work with triggers in a unconventional way, by going through the triggerings and processing their values to achieve the "chopping".

The proper way to implement this is.
With the said 3 steps and work with callbacks.

E.g. every tree has a trigger when the player walks in the tree is registering "Player walked in" through a trigger callback and set's itself as the chopping/action target a variable on the player, when leaving the trigger of the tree, reset the variable to null.

On the player or HUD in your case, when using the function "USE" you just check your current action target for it's type and if you have the right item, if thats the case you call the "action" on your tree. The tree then knows it's own action, which can be adding wood/points or whatever.

Doing it this way does give you the ability to expand on your game easily, you can add new objects and just always call the "action" of the current triggered object, wether it's a tree, stone, or an iron ore in the future. While your USE function stays pretty much the same, but the triggered object then does different things in their "action" function.

Best practice in game programming is to have very generic functions on objects than can be called if from different places in the code. But the object itself knows what to do with it. E.g. you have a base object weapon that implements do shoot, and the player just calls shoot on a mouseclick, but then you have a pistol vs a sniper which do different dmg and maybe other things in their "shoot", they player should not check what weapon they currently use, when they switch they just call another "shoot" on the mouseclick. 
You get where i'm getting at? Tree/Stone -> Trigger with Triggering Callback on EVENT_PHYSICS.TRIGGER_ENTER/EXIT, telling the player they are the "action" object now/or not, and the player just calling on their "action" object in case of a HUD Click. (no for looping through stuff, which also is bad for performance)

With your implementation you also run into trouble because the "triggerings" of the rigidbody is only a helper to notice a trigger leaving, the actual trigger ENTER/EXIT is just a one frame EVENT happening. Like a collision.
